### ISSUE TITLE
Breaking by word a string containing Japanese and Latin characters

### ISSUE THREAD

**Issue by edouard** [twitter/twitter-cldr-rb#260]
**Describe the bug**

We’re using `TwitterCldr::Segmentation::BreakIterator`’s`each_word` method to count words in multiple languages. We just got an exception for a string in Japanese, which contains both Japanese and Latin characters. This is common for when using Western brand names for instance.

**To Reproduce**

``` ruby
string = 'TWITTERド'
iterator = TwitterCldr::Segmentation::BreakIterator.new(:ja)
iterator.each_word(string) {|word| puts word }
#=> /Users/edouard/.rvm/gems/ruby-3.1.2@webtranslateit.com/gems/twitter_cldr-6.11.3/lib/twitter_cldr/segmentation/cj_break_engine.rb:110:in `<': comparison of Integer with nil failed (ArgumentError)
```

Also, this string works:

``` ruby
string = 'WINDYのアカウントを作成する'
iterator = TwitterCldr::Segmentation::BreakIterator.new(:ja)
iterator.each_word(string) {|word| puts word }
#=> WINDY
の
アカウントを作成する
 => #<Enumerator: ...> 
```

Interestingly enough, taking that string above and replacing `WINDY` with `TWITTER` doesn’t work 🤔:

``` ruby
string = 'TWITTERのアカウントを作成する'
iterator = TwitterCldr::Segmentation::BreakIterator.new(:ja)
iterator.each_word(string) {|word| puts word }
#=> /Users/edouard/.rvm/gems/ruby-3.1.2@webtranslateit.com/gems/twitter_cldr-6.11.3/lib/twitter_cldr/segmentation/cj_break_engine.rb:110:in `<': comparison of Integer with nil failed (ArgumentError)
```

**Expected behavior**

The `BreakIterator` shouldn't raise an exception

**Screenshots**

**Environment**
ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-darwin21]

**Additional context**

**Comment by edouard** [comment]
Answering my own questions here...

> Interestingly enough, taking that string above and replacing WINDY with TWITTER doesn’t work 🤔:

``` ruby
string = 'TWITTERのアカウントを作成する'
iterator = TwitterCldr::Segmentation::BreakIterator.new(:ja)
iterator.each_word(string) {|word| puts word }
#=> /Users/edouard/.rvm/gems/ruby-3.1.2@webtranslateit.com/gems/twitter_cldr-6.11.3/lib/twitter_cldr/segmentation/cj_break_engine.rb:110:in `<': comparison of Integer with nil failed (ArgumentError)
```

It seems to be due to the length of the latin word:

``` ruby
string = 'TWITTのアカウントを作成する'
iterator = TwitterCldr::Segmentation::BreakIterator.new(:ja)
iterator.each_word(string) {|word| puts word }
#=> TWITT
の
アカウントを作成する
```

Looks like the error we see has to do with the length of the latin word.

https://github.com/twitter/twitter-cldr-rb/blob/09a1db07bf68b397e482d90290b0bb886ee076e1/lib/twitter_cldr/segmentation/cj_break_engine.rb#L149-L155

**Comment by camertron** [comment]
Hey @edouard, thanks for reporting this. Please see #261 for additional details. The fix has been published in v6.11.4.

**Comment by edouard** [comment]
Cool! Thanks for fixing it so quickly! 👍🏽
